design/fast_finality.md

```diff
@@ -0,0 +1,273 @@
+
+### Production of light batches
+
+A light batch is produced on the required cadence to meet the network's soft-finality window of one second. Only the 
+sequencer produces light batches.
+
+To produce a light batch, the sequencer's host feeds a set of transactions to the enclave. The enclave responds by 
+creating a signed and encrypted *light batch*. This light batch is formally identical to the rollup of the final 
+design, including a list of the provided transactions and a header including information on the current light batch and 
+the hash of the "parent" light batch. This header also includes a monotonically increasing counter, in order to support 
+smart contracts that use `block.number` as a rudimentary clock (e.g. SushiSwap).
+
+The sequencer's host immediately distributes the light batch to all other nodes, who gossip it onwards to other nodes
+(ensuring the sequencer cannot restrict the distribution of light batches to specific nodes, provided one of the nodes 
+who received the light batch is honest). These light batches are not sent to be included on the L1.
+
+When a node receives a light batch, it checks that it has also stored the light batch's parent. If 
+not, it walks the chain backwards, requesting any light batches it is missing until it hits a stored light batch. In 
+the current design, it requests these light batches from random nodes; once a gossip protocol is implemented, it will 
+request the light batches from its known peers.
+
+The linkage of each light batch to its parent also ensures that the sequencer's host cannot feed the enclave a light 
+batch, use RPC requests to gain information about the contents of the corresponding transactions, then feed the enclave 
+a different light batch (e.g. one where the sequencer performs front-running) to be shared with peers. The enclave will 
+automatically reject a second light batch with the same parent.
+
+From the user's perspective, the transactions in the light batch are considered final (e.g. responses to RPC calls from
+the client behave as if the transactions were completely final).
+
+### Production of rollups
+
+Only the sequencer produces rollups. A rollup is produced whenever one of the following conditions is met:
+
+1. The number of transactions across all light-batches since the last rollup exceeds `x`
+2. The total size of all transactions across all light-batches since the last rollup exceeds `y`
+3. The number of blocks since the last rollup was produced exceeds `z`
+
+`x`, `y` and `z` are configurable per network. Rules (1) and (2) ensure that the rollup can fit within the Ethereum gas 
+limit. Rule (3) reduces the risk associated with the sequencer failing.
+
+To produce a rollup, the sequencer's host requests the creation of a rollup. The sequencer's enclave produces a rollup 
+containing all the light batches created since the last rollup, in a sparse Merkle tree structure. This rollup is 
+encrypted and signed, then sent to be included on the L1.
```

Comment:
let's add a bit of thinking here about the boundary.
How will the creation of a rolllup affect the LB chain?
does something reset?